src/ayab/encoders.cpp

```diff
@@ -155,6 +150,38 @@ void Encoders::encA_rising() {
       (hallValue > FILTER_L_MAX[static_cast<uint8_t>(m_machineType)])) {
     m_hallActive = Direction_t::Left;
 
+    // Only set the belt shift the first time a magnet passes the turn mark.
+    // Headed to the right.
+    if (!m_passedLeft && Direction_t::Right == m_direction) {
+      // Belt shift signal only decided in front of hall sensor
+      m_beltShift = digitalRead(ENC_PIN_C) != 0 ? BeltShift::Shifted : BeltShift::Regular;
```

Comment:
Personally I find this line a bit dense, I'd refactor to something like, `if (digital read()) m_beltshift = shifted`.

Reply:
We'd still need to set the BeltShift::Regular, the default is BeltShift::Unknown.

Honestly, the ternary operator is the best option here.

Reply:
I don't care enough to fight this one, happy to let it go :)